src/main/java/org/opentripplanner/middleware/bugsnag/jobs/BugsnagEventHandlingJob.java

```diff
@@ -78,8 +81,13 @@ private void refreshEventRequest(BugsnagEventRequest request) {
         }
         switch (refreshedRequest.status.toLowerCase()) {
             case "completed": {
-                // First, remove "stale" requests (i.e., those that are older than this latest one).
-                Persistence.bugsnagEventRequests.removeFiltered(Filters.lte("dateCreated", request.dateCreated));
+                // First, remove "stale" requests (i.e., those that are older than this latest one) for this project.
+                Persistence.bugsnagEventRequests.removeFiltered(
+                    Filters.and(
+                        lte("dateCreated", request.dateCreated),
+                        eq("projectId", request.projectId)
+                    )
+                );
                 // Next, get and store the new events from the completed request and notify users.
                 List<BugsnagEvent> newEvents = getNewEvents(refreshedRequest);
                 if (newEvents.size() > 0) {
```

Comment:
What if these stale requests cover a different time period? Seems like there is a risk for that with the new approach.

Also, are we replacing the event request in Mongo once it's completed?

Reply:
Sorry I don't think my prior comment was super clear. I think removing the completed request will cause issues with our logic in `BugsnagJobs#getLatestComparisonRequest`. I have some recommended changes in #150.